helsingborg-stad/app-mitt-helsingborg
# Feat/automatic builds on merge cu h546uk

## Explain the changes you’ve made

* Added `ios-build` GitHub Actions workflow file which will build and upload the app to Testflight on pushes to branches with names matching `release/*`.
* Changed Fastlane to build/upload with App Store Connect API Key
* Locked Fastlane XCode version to 12.4
* Add "long upload" to Fastlane which will publish the build and notify testers in the "CD Builds" group

## Explain why these changes are made

The build workflow will automatically run on release commits to reduce developer time needed for manually deployments between commits.

Changing Fastlane to use an App Store Connect API Key also further streamlines the workflow, decoupling the workflow from an individual's account, reduces the amount of environment variables needed, and removes the time-limitation of the previous session-token system.

XCode version is locked to the version provided on the GitHub Actions CI runner to reduce compatibility errors.


## Explain your solution

The iOS build file is a GitHub Actions workflow configuration file that sets-up and runs `fastlane build` on a MacOS GitHub CI runner. In addition to this file, a GitHub environment called "production" has been setup on this repo which contains the environment variables and secrets used.

The environment variables used by the GitHub environment are documented in `ios/fastlane/example.env`.


## How to test the changes?

* Check latest workflow run (https://github.com/helsingborg-stad/app-mitt-helsingborg/actions/runs/817106358)
* Verify no sensitive data is shown
  * Logs are public - GitHub Actions should automatically censor environment secrets as ***
* Verify a build is uploaded on App Store Connect
* Verify the build is working and using production environment (add yourself to the TestFlight "CD Builds" group if needed to test)
  * NOTE: The latest uploaded build "what to test" message says it was built from commit e92b190, however that is a hash from the pull request merge, not the actual commit (which is da2e7a7); builds made from outside a `pull_request` event should show the correct hash id).
  * You can still verify that the hash id used is the same as the HEAD for the workflow run, in the workflow run log under the "Checkout" step:
![image](https://user-images.githubusercontent.com/81566071/117028130-af23e280-acfd-11eb-8971-0af0a1fd3ee8.png)



## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
- [] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.


## Anything else?

The workflow can be run manually once the workflow file exists on the GitHub default branch (develop), but can not currently be run. Previous commits to this branch enabled the build workflow to be run for pull requests for testing, however this is now removed since testing is concluded; check the latest workflow run from the Actions tab in GitHub.
